### DISCUSSION TITLE
FILE UPLOAD FAILS:  /TMP DIR is set to USER:GROUP 1000:1000 with restrictive write permissions

### DISCUSSION THREAD

**Discussion by damanic** [serversideup/docker-php#284] [Q&A] [answered]
Why is it that when I launch a fresh serversideup/php instance my PHP application is unable to upload files?

Surely the /tmp directory should be set up to allow PHP to write?  

Looking at the permissions in the container for the /tmp directory I am seeing

drwxr-xr-x 2 1000 1000 4096

Which means www-data is unable to write.  If I 777 the /tmp dir all is well, but why do I need to do this?

Docker File:

```

# BASE IMAGE INSTALL
FROM serversideup/php:8.0-fpm-apache as base
# Install PHP Imagemagick using regular Ubuntu commands
RUN apt-get update -y \
  && apt-get install -y \
     libxml2-dev \
     php8.0-soap \
     php8.0-imagick \
     php8.0-xdebug \
  && apt-get clean -y \
  && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /usr/share/doc/*

```

I have tried adding

`RUN chown www-data:www-data /tmp && chmod 1777 /tmp`

This fails to chown to the correct UID/GID (remains 1000). But does `1777` which allows access.

Permission shuffle shouldn't be necessary surely, what am I doing wrong?

I am running on Ubuntu (Windows WSL) , which I am predicting will be pointed to as part of the problem.

**Comment by jaydrogers** [comment, marked as answer]
You need to make sure everything matches your host user's id. So if your user in WSL is 1000, then you want to change www-data to 1000 in the container.

See his example: https://github.com/serversideup/spin/blob/main/templates/laravel/Dockerfile

Be sure to use the **beta** images to follow my example with `docker-php-serversideup-set-id`.

**Comment by damanic** [comment]
@jaydrogers That will do it thanks.     I might have found your Dockerfile example if it had comments that described the issue it is trying to solve, and specifically mentioned a common issue running on WSL as an example.
